### PR TITLE
Update xcmsSet to be able to run files with MS1 and MS2

### PR DESCRIPTION
**Pull request by jsaintvanne**
Just adding a filter to keep only MS1 peaks from files containing MS1 and also MSn datas.
The `xdata <- findChromPeaks(raw_data, param=findChromPeaksParam, msLevel=1)` seems not to be an obligation with `msLevel=1`

### CODE REVIEW

File: tools/xcms_xcmsset/xcms_xcmsSet.r [hunk 1]
```diff
@@ -83,14 +83,16 @@ cat("\t\t\tApply filter[s] (if asked)\n")
 if (exists("filterAcquisitionNumParam"))  raw_data <- filterAcquisitionNum(raw_data, filterAcquisitionNumParam[1]:filterAcquisitionNumParam[2])
 if (exists("filterRtParam")) raw_data <- filterRt(raw_data, filterRtParam)
 if (exists("filterMzParam")) raw_data <- filterMz(raw_data, filterMzParam)
+raw_data <- filterMsLevel(raw_data,msLevel=1)
+print(raw_data)
```
Comment (lecorguille) [inline tools/xcms_xcmsset/xcms_xcmsSet.r]: Can you remove this line?

File: tools/xcms_xcmsset/xcms_xcmsSet.r [hunk 2]
```diff
@@ -83,14 +83,16 @@ cat("\t\t\tApply filter[s] (if asked)\n")
 if (exists("filterAcquisitionNumParam"))  raw_data <- filterAcquisitionNum(raw_data, filterAcquisitionNumParam[1]:filterAcquisitionNumParam[2])
 if (exists("filterRtParam")) raw_data <- filterRt(raw_data, filterRtParam)
 if (exists("filterMzParam")) raw_data <- filterMz(raw_data, filterMzParam)
+raw_data <- filterMsLevel(raw_data,msLevel=1)
```
Comment (lecorguille) [inline tools/xcms_xcmsset/xcms_xcmsSet.r]: Do you know if it's time consuming?
If yes, can we detect if there are other level before filtering?

File: tools/xcms_xcmsset/xcms_xcmsSet.r (hunk 1 above)
Comment (jsaintvanne) [reply tools/xcms_xcmsset/xcms_xcmsSet.r]: Removed !

File: tools/xcms_xcmsset/xcms_xcmsSet.r (hunk 2 above)
Comment (jsaintvanne) [reply tools/xcms_xcmsset/xcms_xcmsSet.r]: I tried to run it on two files : first with only MS1, second with MS1 and MS2 : 
`time ./tools/xcms_xcmsset/xcms_xcmsSet.r image '/home0/jsaintvanne/galaxy/database/files/002/dataset_2841.dat'  BPPARAM ${GALAXY_SLOTS:-1}   method CentWave   ppm 25 peakwidth "c(5,30)"  snthresh 10 prefilter "c(3,100)" mzCenterFun wMean integrate 1 mzdiff -0.001 fitgauss FALSE noise 0 verboseColumns FALSE
Creating a new generic function for ‘smooth’ in package ‘MSnbase’
Creating a new generic function for ‘trimws’ in package ‘MSnbase’
in method for ‘coerce’ with signature ‘"IBSpectra","MSnSet"’: no definition for class “IBSpectra”
Detecting mass traces at 25 ppm ... OK
Detecting chromatographic peaks in 3962 regions of interest ... OK: 78 found.
Note: you might want to set/adjust the 'sampclass' of the returned xcmSet object before proceeding with the analysis.

real	0m19.151s
user	0m17.044s
sys	0m0.344s
`
`time ./tools/xcms_xcmsset/xcms_xcmsSet.r image '/home0/jsaintvanne/galaxy/database/files/002/dataset_2840.dat'  BPPARAM ${GALAXY_SLOTS:-1}   method CentWave   ppm 25 peakwidth "c(5,30)"  snthresh 10 prefilter "c(3,100)" mzCenterFun wMean integrate 1 mzdiff -0.001 fitgauss FALSE noise 0 verboseColumns FALSE
Creating a new generic function for ‘smooth’ in package ‘MSnbase’
Creating a new generic function for ‘trimws’ in package ‘MSnbase’
in method for ‘coerce’ with signature ‘"IBSpectra","MSnSet"’: no definition for class “IBSpectra”
Detecting mass traces at 25 ppm ... OK
Detecting chromatographic peaks in 3962 regions of interest ... OK: 78 found.
Note: you might want to set/adjust the 'sampclass' of the returned xcmSet object before proceeding with the analysis.

real	0m24.416s
user	0m17.704s
sys	0m0.512s
`

So maybe we can try to detect if there is only MS or more in input files ?

**Comment by jsaintvanne** [comment]
I tested the time function with MS file and MS+MSMS file with and without this 
```#Apply this filter only if file contain MS and MSn
if(length(unique(msLevel(raw_data)))!= 1){
	raw_data <- filterMsLevel(raw_data,msLevel=1)
}```

It is the following results : 
```
MS+MSMS file
real	0m20.133s
user	0m17.344s
sys	0m0.360s

MS file with filter
real	0m18.220s
user	0m17.164s
sys	0m0.248s

MS without filter
real	0m17.977s
user	0m17.008s
sys	0m0.208s
```
It looks that it doesn't change a lot the time...

**Comment by lecorguille** [comment]
Cool!

**Pull request merged**